### PR TITLE
update task

### PR DESCRIPTION
**Pull request by ArrogantPeter**

### CODE REVIEW

**Review by hozakowskimilosz — APPROVED** [review]

File: packages/functions/src/lambda/getByColumnId/lambda.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { APIGatewayProxyEventV2 } from "aws-lambda";
+import { ApiError } from "src/model/errors";
+
+
+const taskRepository = TaskRepository.getTaskRepository();
+
+export async function main(e: APIGatewayProxyEventV2) {
+
+    try{    
+        const columnId = e.queryStringParameters.columnId;
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/lambda/getByColumnId/lambda.ts:12]: image
WERE ZOD?

File: packages/functions/src/lambda/task/update/lambda.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const taskRepository = TaskRepository.getTaskRepository();
+
+export async function main(e: APIGatewayProxyEventV2) {
+    const body=JSON.parse(e.body??"")
+    try {
+        const itemCount = taskRepository.getById(body.id)
+        if (!itemCount) return ApiResponse.notFound(`Task with id ${body.id} was not found!`);
+
+        const task = TaskSchema.parse(body)
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/lambda/task/update/lambda.ts:18]: parse on beginning of main function

File: packages/functions/src/lambda/task/update/lambda.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        await taskRepository.put(task);
+        const res = "Updated task."
+        return ApiResponse.ok(res)
+    }catch(err){
+        if (err instanceof z.ZodError) {
+            const res = err.issues.map(e=>`${e.message} at field ${e.path}`)
+            return ApiResponse.notFound(res);
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/lambda/task/update/lambda.ts:25]: Not found? 400 for me: )

File: packages/functions/src/repositories/taskRepository.ts [hunk 4]
```diff
@@ -19,7 +19,8 @@ export default class TaskRepository{
         const params = {
             TableName: Table.Tasks.tableName,
         };
-        const result = (await this.dynamoDb.scan(params).promise()).Items?.map(e=>TaskSchema.parse(e) as Task)
+        const dbResponse = await this.dynamoDb.scan(params).promise()
+        const result = dbResponse.Items?.map(e=>TaskSchema.parse(e) as Task)
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/repositories/taskRepository.ts:23]: you can inline that 
return dbResponse.Items?.map(e=>TaskSchema.parse(e) as Task) ?? []

**Review by hozakowskimilosz — APPROVED** [review]

File: packages/functions/src/repositories/taskRepository.ts [hunk 5]
```diff
@@ -37,14 +38,29 @@ export default class TaskRepository{
...
-    public async add(task : Task): Promise<void>{ 
+    public async put(task : Task): Promise<Task | void>{ 
         const params = {
             TableName: Table.Tasks.tableName,
             Item:task,
         };
         await this.dynamoDb.put(params).promise();
     }
 
+    public async getByColumnId(columnId: number): Promise<Task[] | undefined > {
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/repositories/taskRepository.ts:49]: do not return undefined here

File: packages/functions/src/repositories/taskRepository.ts [hunk 6]
```diff
@@ -37,14 +38,29 @@ export default class TaskRepository{
...
+        const params = {
+            TableName: Table.Tasks.tableName,
+            FilterExpression: "columnId = :columnId",
+            ExpressionAttributeValues: {
+                ":columnId" : columnId,
+            }
+        }
+        const dbResponse = await this.dynamoDb.scan(params).promise()
+        if(dbResponse.Count==0) return undefined;
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/repositories/taskRepository.ts:58]: remove it

File: stacks/AppStack.ts [hunk 7]
```diff
@@ -17,6 +17,8 @@ function createApi(stack: Stack) {
       "POST /task/add": "packages/functions/src/lambda/task/add/lambda.main",
       "GET /task/get-all": "packages/functions/src/lambda/task/get-all/lambda.main",
       "DELETE /task/delete": "packages/functions/src/lambda/task/delete/lambda.main",
+      "GET /task/getByColumnId": "packages/functions/src/lambda/task/getByColumnId/lambda.main",
+      "POST /task/update": "packages/functions/src/lambda/task/update/lambda.main"
```
Comment (jan-kowalczyk-wttech) [inline stacks/AppStack.ts:21]: To update something we should use PUT :)

File: packages/functions/src/repositories/taskRepository.ts [hunk 8]
```diff
@@ -37,14 +38,29 @@ export default class TaskRepository{
         return result
     }
 
-    public async add(task : Task): Promise<void>{ 
+    public async put(task : Task): Promise<Task | void>{
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/repositories/taskRepository.ts:41]: why void or task? Something is wrong here

File: packages/functions/src/repositories/taskRepository.ts [hunk 9]
```diff
@@ -19,8 +19,9 @@ export default class TaskRepository{
         const params = {
             TableName: Table.Tasks.tableName,
         };
-        const result = (await this.dynamoDb.scan(params).promise()).Items?.map(e=>TaskSchema.parse(e) as Task)
-        return result ?? [];
+        const dbResponse = await this.dynamoDb.scan(params).promise()
+        const result = dbResponse.Items?.map(e=>TaskSchema.parse(e) as Task) ?? []
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/repositories/taskRepository.ts:23]: inline, you don't need to put repsonse to variable here

File: packages/functions/src/lambda/task/update/lambda.test.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        //WHEN
+        const result = await main(event)
+
+        // THEN
+        expect(result?.statusCode ?? 0).toBe(400)
+    })
+
+    test("should return status code 500", async () => {
+        // GIVEN
+        vi.spyOn(TaskRepository.prototype, "put").mockRejectedValue(undefined)
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/lambda/task/update/lambda.test.ts:64]: What is returning database on invalid request? Did you check that? Probably will reject with Error

File: packages/functions/src/lambda/task/update/lambda.test.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+        const event: APIGatewayProxyEventV2 = {
+            body: JSON.stringify(body)
+        } as any
+
+        //WHEN
+        const result = await main(event)
+
+        // THEN
+        expect(result?.statusCode ?? 0).toBe(400)
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/lambda/task/update/lambda.test.ts:59]: why ??, this is test, you don't need to check if statusCode is defined

File: packages/functions/src/lambda/task/update/lambda.test.ts [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+        // WHEN
+        const result = await main(event)
+
+        // THEN
+        expect(result?.statusCode ?? 0).toBe(200)
+        expect(JSON.parse(result?.body ?? "")).toBe(`Updated task.`)
+    })
+
+    test.each([
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/lambda/task/update/lambda.test.ts:34]: nice 👍

File: packages/functions/src/lambda/task/update/lambda.test.ts [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+        const event: APIGatewayProxyEventV2 = {
+            body: JSON.stringify(taskMock)
+        } as any
+
+        // WHEN
+        const result = await main(event)
+
+        // THEN
+        expect(result?.statusCode ?? 0).toBe(200)
+        expect(JSON.parse(result?.body ?? "")).toBe(`Updated task.`)
```
Comment (jan-kowalczyk-wttech) [inline packages/functions/src/lambda/task/update/lambda.test.ts:31]: same here

File: packages/functions/src/lambda/task/update/lambda.test.ts (hunk 10 above)
Comment (ArrogantPeter) [reply packages/functions/src/lambda/task/update/lambda.test.ts:64]: documentation doesn't really tell us a lot about possible errors and it is quite hard to fail requests to dynamoDB and it is already quite foolproof

**Pull request merged**